grades/api.py

```diff
@@ -230,3 +232,31 @@ def generate_program_certificate(user, program):
         user.username,
         program.title
     )
+
+
+def update_or_create_combined_final_grade(user, course):
+    """
+    Update or create CombinedFinalGrade
+
+    Args:
+        user (User): a django User
+        course (Course): a course model object
+    """
+    mmtrack = get_mmtrack(user, course.program)
```

Comment:
Should we also be checking `has_exam` here so we don't bother checking for a course that doesn't have one?

Reply:
Its is called from signals where we save a an exam grade, so the course should have `has_exam` true. And it is called in the management command where we check if the course has exam. Should I check that inside the function anyway?

Reply:
It mostly occurred to me because I saw that it's also called on `post_save` for `FinalGrade` and that we could be executing this against courses in programs other than DEDP. That said, those will never have `ProtoredExamGrade`s created for them, so at worst we're just doing a few unnecessary lookups by not checking for `has_exam`. I don't think there's a functional difference aside from that overhead though so I'm fine either way.

Reply:
right, I forgot about the final grade signal